debuginfo/src/dwarf.rs

```diff
@@ -643,18 +647,21 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
 
             let name = match symbol_name {
                 Some(name) => Some(name),
-                None => self.inner.resolve_function_name(entry)?,
+                None => self.inner.resolve_function_name(entry).ok().flatten(),
             };
 
             // Avoid constant allocations by collecting repeatedly into the same buffer and
             // draining the results out of it. This keeps the original buffer allocated and
             // allows for a single allocation per call to `resolve_lines`.
-            let lines = self.resolve_lines(&range_buf)?;
+            let lines = self.resolve_lines(&range_buf);
 
             let function = Function {
                 address: function_address,
                 size: function_size,
-                name: Name::with_language(name.unwrap_or_default(), self.language),
+                name: Name::with_language(
+                    name.unwrap_or_else(|| Cow::from("<name omitted>")),
```

Comment:
Could you leave this an empty string and coerce in the caller? We rely on this being empty in Sentry since we show our own placeholders which have semantic meaning.